app/models/profile.rb

```diff
@@ -37,4 +32,26 @@ def self.static_fields
   def clear!
     update(data: {})
   end
+
+  # Lazily add accessors for profile fields on first use
+  def method_missing(method_name, *args, **kwargs, &block)
+    match = method_name.match(ATTRIBUTE_NAME_REGEX)
+    field = ProfileField.find_by(attribute_name: match[:attribute_name])
```

Comment:
Are there any concerns over profile field names containing non-`\w` characters? I mean, I personally have them, but are they reasonable? :joy:

Specifically, this concern is coming from the fact that it [doesn't look like we enforce this in profile fields](https://github.com/forem/forem/blob/877cb5c3222dc57263dc611cce21e0463b630b22/app/models/profile_field.rb#L21-L26) and Ruby allows non-word characters in a method name:

<img width="652" alt="image" src="https://user-images.githubusercontent.com/108205/131902428-75fc6ac9-fdcf-4035-a5e4-6a0f917b7f0e.png">

They can even have spaces:

<img width="578" alt="image" src="https://user-images.githubusercontent.com/108205/131904333-de0624ad-a13d-4e4e-a547-e8456a216e2b.png">


So even if the profile field's name does not match `\A\w+=?\z` we could probably still use it.

Another weird thing is that [we don't even have a guarantee that our DB column names only contain `\w` characters](https://github.com/forem/forem/blob/877cb5c3222dc57263dc611cce21e0463b630b22/db/schema.rb#L932).

Reply:
This is a good point. The `attribute_name` is autogenerated from the `Label`. In this process, only `\w` characters are considered and underscores are added. So the "omg lol" would actually be turned into `omg_lol` and "omg_lol=` accessors. Labels that don't contain any sensible characters are a bit of an issue though, as they would get created with `attribute_name`s of "". I think I'd like to fix this in a follow-up PR though as it's somewhat orthogonal to the main purpose of this PR.

> Another weird thing is that we don't even have a guarantee that our DB column names only contain \w characters.

😱 AR automatically adds predicate methods for boolean columns anyway, so this is definitely not necessary.